docs/source/routing/observability/debugging-client-requests.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: Debugging Client Requests to GraphOS Router
+subtitle: Options for analyzing and debugging incoming requests
+description: Learn how to use GraphOS router telemetry and GraphOS Insights to inspect and debug incoming HTTP client requests.
+---
+
+By default, the GraphOS Router operates without generating HTTP request logs or exporting telemetry metrics.  This minimizes potentially high observability costs that can result from high request volumes.
+
+Without enabling additional telemetry, the easiest way to analyze GraphQL operation data is through [GraphOS Insights](/graphos/platform/insights). Once you've [connected your router to your GraphOS account](/graphos/platform/insights/sending-operation-metrics#from-the-apollo-router-or-apollo-server), it automatically reports operation and field usage metrics. You can view these metrics in GraphOS Studio.
+
+If you need more data than what GraphOS provides by default, you can configure your router to collect and export additional telemetry.
+
+## Using GraphOS Insights
+
+GraphOS Studio lets you analyze data from failed requests, such as GraphQL error messages ([if enabled](/graphos/routing/graphos-reporting#errors)) and the ID of the client making the request. You can also [segment your insights data](/graphos/platform/insights/client-segmentation) based on the client ID.
+
+<Tip>
+
+[Learn how to ensure client IDs are included in all requests.](/graphos/routing/observability/client-id-enforcement)
+
+</Tip>
+
+## Enabling additional telemetry
+
+Use the methods below if you need information outside of what's presented in GraphOS Studio to debug client requests.
+
+<Note>
+
+If you want to debug client requests in your own environment, Apollo recommends first doing so in a non-production environment or using logic to debug on a per-request basis.
+
+</Note>
+
+### Telemetry spans and traces
+
+Once you enable [router telemetry](/graphos/routing/observability/telemetry), traces have spans labeled `router`. These spans contain the HTTP information for a given client request via [OpenTelemetry standard attributes](/graphos/reference/router/telemetry/instrumentation/standard-attributes#router) such as `error.type` and `http.response.status_code`.
+
+These attributes include helpful information about the request and response. Spans, however, don't include an attribute for the HTTP body text. If you want to see the GraphQL operation, first check your [GraphOS Insights](/graphos/platform/insights) or use the other options listed below.
+
+### Request and response logging
+
+You can conditionally include the request body in your logs using the [`experimental-when-header`](/graphos/reference/router/telemetry/log-exporters/overview#requestresponse-logging) configuration option.
+
+```yaml title="router.yaml"
+telemetry:
+  exporters:
+    logging:
+      # If one of these headers matches we will log supergraph and subgraphs requests/responses
+      experimental_when_header:
```

Comment:
Let's change this to use the event mechanism. `experimental_when_header` is actually going away https://github.com/apollographql/router/pull/6285

Users should use:
```
telemetry:
  instrumentation:
    spans:
      default_attribute_requirement_level: recommended
      mode: spec_compliant
    events:
      supergraph:
        "resp":
          level: info
          attributes:
            graphql.document: true
            response:
              response_data: $.*
          on: event_response
          message: "logged request"
```

Reply:
Thanks @BrynCooke , GTK.

I'm a little confused about this config. If the intent is to log the GraphQL operation, would it make more sense to use `telemetry.instrumentation.events.supergraph.request`?

E.g.

```
telemetry:
  instrumentation:
    spans:
      default_attribute_requirement_level: recommended
      mode: spec_compliant
    events:
      supergraph:
        request:
          level: info
          attributes:
            graphql.document: true
          on: event_request
          message: "logged request"
```

Reply:
@BrynCooke , could you take a review this code sample to see if it makes sense?

Reply:
Thanks @devfreddy-apollo . From the [conversation in the open PR](https://github.com/apollographql/router/pull/6563#discussion_r1943611841), do you think this is still worth documenting given it's not conditionally done? (Which I believe is why @smyrick  was documenting it in the first place.)